src/Forte.Optimizely.ContentUsage/Api/Features/ContentUsage/ContentUsageController.cs

```diff
@@ -31,24 +31,49 @@ public ContentUsageController(IContentTypeRepository contentTypeRepository, Cont
     [SwaggerResponse(StatusCodes.Status200OK, null, typeof(IEnumerable<ContentUsageDto>))]
     [SwaggerResponse(StatusCodes.Status404NotFound)]
     [Route("[action]", Name = GetContentUsagesRouteName)]
-    public ActionResult GetContentUsages([FromQuery] [BindRequired] Guid guid)
+    public ActionResult GetContentUsages([FromQuery] GetContentUsagesQuery queryData)
     {
-        var contentType = _contentTypeRepository.Load(guid);
+        var contentType = _contentTypeRepository.Load(queryData.Guid);
 
         if (contentType == null) return NotFound();
 
-        var contentUsages = _contentUsageService.GetContentUsages(contentType);
+        var contentUsagesQuery = _contentUsageService.GetContentUsages(contentType);
 
-        var contentUsagesDto = contentUsages.Select(contentUsage => new ContentUsageDto
+        contentUsagesQuery = string.IsNullOrEmpty(queryData.Query)
+            ? contentUsagesQuery
+            : contentUsagesQuery.Where(x => x.Name.Contains(queryData.Query, StringComparison.InvariantCultureIgnoreCase));
+        
+        var contentUsages = contentUsagesQuery.ToArray();
+        
+        var currentPage = queryData.Page <= 0 ? 0 : queryData.Page - 1;
+        
+        const int itemsPerPage = 25;
+        var contentUsagesDto = contentUsages
+            .Sort(queryData)
+            .Skip(currentPage * itemsPerPage)
+            .Take(itemsPerPage)
+            .Select(contentUsage => new ContentUsageDto
         {
             Id = contentUsage.ContentLink.ID,
-            ContentTypeGuid = guid,
+            ContentTypeGuid = queryData.Guid,
             Name = contentUsage.Name,
             LanguageBranch = contentUsage.LanguageBranch,
             PageUrls = _contentUsageService.GetPageUrls(contentUsage),
             EditUrl = _contentUsageService.GetEditUrl(contentUsage)
         });
 
-        return Ok(contentUsagesDto);
+        var totalPages = contentUsages.Count() / itemsPerPage;
+        return Ok(new GetContentUsagesResponse
+        {
+            ContentUsages = contentUsagesDto,
+            TotalPages = totalPages
```

Comment:
Consider renaming `TotalPages` to `PageCount`.
Since it stores a number, not a collection of pages ;)

Reply:
According to https://dictionary.cambridge.org/dictionary/english/total I think it's a good name.
What do you think?

Reply:
`total` is fine, yet personally not a fan of ending a name with a plural noun when it's not enumerable.

There is a guideline:
> - DO name collection properties with a plural phrase describing the items in the collection instead of using a singular phrase followed by "List" or "Collection".
so I would keep `TotalPages` name for a type like `IEnumerable<TotalPage>`.

Since that is just a preference, it is up to you.

Reply:
Yes, but:

1. it's not a collection
2. TotalPage isn't also correct in context of pages

I'll leave this as it is.